src/main/java/com/artipie/api/RepositoryName.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * The MIT License (MIT) Copyright (c) 2020-2021 artipie.com
+ * https://github.com/artipie/artipie/LICENSE.txt
+ */
+
+package com.artipie.api;
+
+/**
+ * Repository name.
+ * @since 0.26
+ */
+public interface RepositoryName {
```

Comment:
I'd suggest to add implementation `FromRequest` that will accept `RoutingContext` and layout into ctor and obtain repository name in `string()` method with the logic to obtain repo name from either one request parameter from two.